yt_dlp/networking/_curlcffi.py

```diff
@@ -27,11 +28,19 @@
 if curl_cffi is None:
     raise ImportError('curl_cffi is not installed')
 
-curl_cffi_version = tuple(int_or_none(x, default=0) for x in curl_cffi.__version__.split('.'))
 
-if curl_cffi_version != (0, 5, 10):
+def group_or_none(n):
+    r = re.match(r'\d+', n)
+    if r:
+        return r.group()
+
+
+curl_cffi_version = tuple(int_or_none(group_or_none(n), default=0) for n in curl_cffi.__version__.split('.'))
```

Comment:
If the input string was not numeric, a match object with a pattern of `r'(\d+)?'` will return an empty string when `group()` is called, then `int_or_none` will default to `0`
```suggestion
curl_cffi_version = tuple(int_or_none(re.match(r'(\d+)?', n).group(), default=0) for n in curl_cffi.__version__.split('.'))
```